[Quant][PT2E]Make _fuse_conv_bn_ support graph capture by torch._dynamo.export

leslie-fang-intel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #107951
* #106958
* #106838
* #106836

**Summary**
The latest check-in https://github.com/pytorch/pytorch/commit/a0cfaf068887a608c00de4f5ceaf30242c6af720 for the conv-bn folding assumes the graph is captured by the new graph capture API `torch._export.capture_pre_autograd_graph`. Since we still need to use the original graph capture API `torch._dynamo_export` in 2.1 release. So, this check-in made negative impact to workloads' performance heavily. Made this PR to fix this issue by trying to make the conv-bn folding function workable with both new and original graph capture API.

leslie-fang-intel committed:
Make _fuse_conv_bn_ support graph capture by torch._dynamo.export

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E]Make _fuse_conv_bn_ support graph capture by torch._dynamo.export"




[ghstack-poisoned]

leslie-fang-intel commented:
@jgong5 @jerryzh168 Please kindly help to take a look of this PR.

jerryzh168 commented on torch/ao/quantization/pt2e/utils.py:
```diff
@@ -47,6 +47,9 @@ def fold_bn_weights_into_conv_node(
     conv_w = _get_tensor_constant_from_node(conv_weight_node, m)
     conv_b = _get_tensor_constant_from_node(conv_bias_node, m)
     transpose = not (conv_node.target == torch.ops.aten.conv2d.default)
+    if conv_node.target == torch.ops.aten.convolution.default:
+        assert type(conv_node.args[6]) is bool
+        transpose = conv_node.args[6]
```
could you add a TODO to remove these

leslie-fang-intel replied:
Thanks for the comments, TODO added here and others in this diff.

leslie-fang-intel committed:
Update on "[Quant][PT2E]Make _fuse_conv_bn_ support graph capture by torch._dynamo.export"


**Summary**
The latest check-in https://github.com/pytorch/pytorch/commit/a0cfaf068887a608c00de4f5ceaf30242c6af720 for the conv-bn folding assumes the graph is captured by the new graph capture API `torch._export.capture_pre_autograd_graph`. Since we still need to use the original graph capture API `torch._dynamo_export` in 2.1 release. So, this check-in made negative impact to workloads' performance heavily. Made this PR to fix this issue by trying to make the conv-bn folding function workable with both new and original graph capture API.

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E]Make _fuse_conv_bn_ support graph capture by torch._dynamo.export"


**Summary**
The latest check-in https://github.com/pytorch/pytorch/commit/a0cfaf068887a608c00de4f5ceaf30242c6af720 for the conv-bn folding assumes the graph is captured by the new graph capture API `torch._export.capture_pre_autograd_graph`. Since we still need to use the original graph capture API `torch._dynamo_export` in 2.1 release. So, this check-in made negative impact to workloads' performance heavily. Made this PR to fix this issue by trying to make the conv-bn folding function workable with both new and original graph capture API.

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][PT2E]Make _fuse_conv_bn_ support graph capture by torch._dynamo.export"


**Summary**
The latest check-in https://github.com/pytorch/pytorch/commit/a0cfaf068887a608c00de4f5ceaf30242c6af720 for the conv-bn folding assumes the graph is captured by the new graph capture API `torch._export.capture_pre_autograd_graph`. Since we still need to use the original graph capture API `torch._dynamo_export` in 2.1 release. So, this check-in made negative impact to workloads' performance heavily. Made this PR to fix this issue by trying to make the conv-bn folding function workable with both new and original graph capture API.

[ghstack-poisoned]

leslie-fang-intel commented:
@pytorchbot merge
